core/src/main/scala/kafka/log/Log.scala

```diff
@@ -755,8 +755,10 @@ class Log(@volatile private var _dir: File,
           producerStateManager.updateParentDir(dir)
           // re-initialize leader epoch cache so that LeaderEpochCheckpointFile.checkpoint can correctly reference
           // the checkpoint file in renamed log directory
-          initializeLeaderEpochCache()
-          initializePartitionMetadata()
+          if (reinitialize) {
```

Comment:
nit: "shouldReinitialize" might be better name for a boolean var.

Reply:
Thanks for the suggestion. I've renamed the variable.